### DISCUSSION TITLE
action_cable example Noticed 2

### DISCUSSION THREAD

**Discussion by stap780** [excid3/noticed#424] [Q&A]
Hello!
Spent several days with Noticed 2.0 and still can't configure setup for action_cable. Maybe someone have example? Or i miss something in docs?
Standard scenario: User need to have notice about different events (create export file, create pdf file and etc).
I add ExportNotifier and also have index (Notification) with turbo_frame and turbo_stream (with current_user).
I am also add authentication from action_cable docs.

Tried create ExportNotifier from console. And all good but notification not appear in list on index page (only after page reload).
I took basic setup:
deliver_by :action_cable do |config|
  config.stream = ->{ recipient }
  config.message = ->{ params.merge( user_id: recipient.id) }
end

If i add "  config.channel = "Noticed::NotificationsChannel" " than have error about - unitialized constand Noticed::NotificationsChannel

Any help will be good :)

**Comment by stap780** [comment]
Hi!
I did it and configured out :) If some one will have a questions than i wrote small instruction for myself and i can share it.

**Reply by joshkanna** [reply]
Could you please explain how you solved this?  I am so lost as I can't figure out action cable delivery method configuration!

**Reply by stap780** [reply]
https://www.ketago.com/blog/rails-7-notification-with-noticed-2-and-turbo-stream

**Reply by tkhobbes** [reply]
I am struggling. It gets up to the actual turbo stream delivery method, but then I get an error:
ActionView::Template::Error
unknown local: :notification

I think it has to do with the code in the turbo_stream.rb delivery method:
`notification.update_public_notifications`

(my method is called update_public_notifications, as opposed to broadcast_update_to_bell)...

**Reply by hdaniel** [reply]
@stap780 Thanks for that write up!  I found it useful.
